src/CoreBundle/private/library/classes/TCMSFields/TCMSFieldExtendedLookup.class.php

```diff
@@ -152,6 +150,30 @@ protected function _GetOpenWindowJS(&$oPopupTableConf)
         return $js;
     }
 
+    private function getTargetListRestriction(): array
+    {
+        $restrictionExpression = $this->oDefinition->GetFieldtypeConfigKey('restriction');
+        if (null === $restrictionExpression) {
+            return [];
+        }
+        if (false === \strpos($restrictionExpression, '=')) {
+            return [];
+        }
+        $equalPosition = \mb_strpos($restrictionExpression, '=');
+        $restrictionField = trim(\mb_substr($restrictionExpression, 0, $equalPosition));
+        $restrictionValue = trim(\mb_substr($restrictionExpression, $equalPosition+1));
+        $restrictionValue = mb_substr($restrictionValue, 2, -2); // ignore [{}]
```

Comment:
Is this the line where something was fixed (I can't really easily tell when comparing old to new code)?

This cuts four characters regardless of their content? Maybe check first?

Reply:
There is no single line fix. The problem was, that the old code would take the restriction:

`foo=[{bar}]`

at face value. That is, it would use `foo` as the restriction field and `[{bar}]` as the restriction value. Instead, the content of [{bar}] is to be used - and to be replaced by a field value IF bar is a field name.

It may make sense to change the definition in suche a way, that `[{}]` will always assume that we are dealing with a field while any other content will be taken as a literal value.

so that `foo=[{bar}]` would mean
restrictionField = foo
restrictionValue = content of field bar

and `foo=bar` would mean
restrictionField = foo
restrictionValue = bar

But if we decided to go that way we also need to check the lookup field. That may exceed the scope of a patch release...

I would feel better though if the logic used in the lookup class was identical to the logic used here... so maybe we do need to revisit this code section

Reply:
refactored to allow static string restrictions as well as field references - and updated field documentation to match what the field now does.